### ISSUE TITLE
Scenery objects not showing up and unable to select using the object picker tool

### ISSUE THREAD

**Issue by astraylife1** [OpenRCT2/OpenRCT2#21552]
### Operating System

MacOS

### OpenRCT2 build

OpenRCT2, v0.4.9 (a0b1868 on develop)

### Base game

RollerCoaster Tycoon 2

### Describe the issue

I have noticed that certain custom scenery objects are not showing up in the scenery selection menu, even when I use the search function. The object could even exist in the park already, and by trying to use the object picker eyedropper tool to select the object, the game ignores it as if I'm clicking on nothing.

To workaround the issue, I have found that by opening up the object selection and locating the scenery object, I find it is already selected as it should be, and then closing the window, the object now shows up under the scenery selection and can be selected using the picker. But every time I reopen the save file, I have to repeat the process.

### Steps to reproduce

Open the attached park
Try and select the little red walls with the eye dropper tool. You can't
Try and find object titled Short Rock Wall Paintabul in the all scenery section in the scenery selection. You can't
Go into object selection, under advanced, walls, you can search for "paint" and the wall shows up. Close window.
Now the object is suddenly showing up in scenery and selectable.

### Attachments

[test.park.zip](https://github.com/OpenRCT2/OpenRCT2/files/14541958/test.park.zip)

**Comment by astraylife1** [comment]
I noticed that if I simply open the object selection window and then close it, the objects then show up in the scenery manager and can be selected by the eye dropper tool. Not sure why this is occurring but I do know at least one other user has reported the same issue.

**Comment by Paulicus25** [comment]
I also have this problem. I downloaded OpenRCT2-0.4.9-develop-5a70fd9-windows-portable-x64.zip today (20 March 2024) to see if it has been fixed. First noticed around the time of the launch of 0.4.9. As you can see I use windows (specifically windows 10) so it is not a mac issue. I have this issue on every scenario I have tried recently (working through added attractions). Always add in a bunch of custom scenery objects for a more enjoyable time. 

Exactly as OP states. Still broken. My current scenario is RCT1 scenario Roman Village if you need to know, but I've had it on the last few I've played. When loading the save game, the custom scenery is unavailable until the window is opened.

It actually affects game play. Every time you open the object selection window to fix the scenery availability it restarts your current research progress. As such players who want to play somewhat normally but with custom scenery are now restricted to waiting until just after a research completes before they can save their game, either that or get delayed research results, or go without the scenery on save load until one research completes. Very frustrating.

**Comment by simonrshepherd** [comment]
+1 for this.

Some more information for devs, if it helps.

I can see an object, in this case '00000002|FSCRBOU4|AE8CE08F', loaded into the park in large_scenery slot 553. It renders on the map. However, the object (Fisch rocks 2x2) is not available in the scenery window.

Using a plugin, I can remove the object from that slot, and re-add it back in. But that doesn't help.

If I remove it from that slot, of course it disappears from the map. If I then re-select the object using the object selection screen, it places it into a different slot. Now, it's listed correctly in the scenery window. But, now all the pre-existing placements are using the wrong slot reference, of course.

I've seen this on a couple parks. I don't know if multiplayer might be the trigger - perhaps @astraylife1 or @Paulicus25 could shed light on if their files recently went through a multiplayer server as mine did.

**Comment by Paulicus25** [comment]
No, nothing to do with multiplayer. I always play single player. All it requires simply saving and loading any save file AT ALL with certain items in the selected in the object selection. That is all. I have this problem in every single game i play right now. Clearly the devs know about it, as seemingly identical/combined problem #21496 is currently marked in milestones for 0.4.12, although the deadline for it is seemingly only 2 days away so whether it actually gets worked on and fixed is another story. I hope it does tho, as it causes me much frustration.

**Comment by FeloniousKush** [comment]
Experiencing the same issue on a fresh install of OpenRCT2 on Mac OS in a new multiplayer server. Interestingly enough this didn't occur on a Windows client connecting to the very same server.

**Comment by simonrshepherd** [comment]
@FeloniousKush enable sandbox mode and this should stop happening!
